pydantic/_internal/_typing_extra.py

```diff
@@ -80,6 +85,10 @@ def is_literal_type(type_: type[Any]) -> bool:
     return Literal is not None and get_origin(type_) in LITERAL_TYPES
 
 
+def is_deprecated_instance(instance: Any) -> bool:
+    return isinstance(instance, tuple(DEPRECATED_TYPES))
```

Comment:
I think we should make `DEPRECATED_TYPES` a tuple above instead of a set, so we don't need to create a new tuple every time this function is called.